src/main/java/de/tum/in/www1/artemis/config/migration/entries/MigrationEntry20240103_143700.java

```diff
@@ -0,0 +1,500 @@
+        log.info("Migrated {}/{} {} participations ({}%)", finishedCounter.get(), totalCount, migrationType, String.format("%.2f", percentage));
+        log.info("Estimated time remaining: {} for {} repositories",
+                TimeLogUtil.formatDuration(getRestDurationInSeconds(finishedCounter.get(), totalCount, reposPerEntry, threadCount)), migrationType);
+    }
+
+    private long getRestDurationInSeconds(final long done, final long total, final long reposPerEntry, final long threads) {
+        final long stillTodo = total - done;
+        final long timePerEntry = ESTIMATED_TIME_PER_REPOSITORY * reposPerEntry;
+        return (stillTodo * timePerEntry) / threads;
+    }
+
+    private String migrateTestRepo(ProgrammingExercise programmingExercise) throws URISyntaxException {
+        return cloneRepositoryFromBitbucketAndMoveToLocalVCS(programmingExercise, programmingExercise.getTestRepositoryUri());
+    }
+
+    /**
+     * Migrate auxiliary repositories of the given programming exercise.
+     *
+     * @param programmingExercise the programming exercise to migrate the auxiliary repositories for
+     */
+    private void migrateAuxiliaryRepositories(ProgrammingExercise programmingExercise) {
+        for (var repo : getAuxiliaryRepositories(programmingExercise.getId())) {
+            try {
+                if (repo.getRepositoryUri() == null) {
+                    log.error("Repository URI is null for auxiliary repository with id {}, cant migrate", repo.getId());
+                    continue;
+                }
+                var url = cloneRepositoryFromBitbucketAndMoveToLocalVCS(programmingExercise, repo.getRepositoryUri());
+                if (url == null) {
+                    log.error("Failed to migrate auxiliary repository for programming exercise with id {}, keeping the url in the database", programmingExercise.getId());
+                }
+                else {
+                    log.debug("Migrated auxiliary repository with id {} to {}", repo.getId(), url);
+                    repo.setRepositoryUri(url);
+                    auxiliaryRepositoryRepository.save(repo);
+                }
+            }
+            catch (Exception e) {
+                log.error("Failed to migrate auxiliary repository with id {}", repo.getId(), e);
+            }
+        }
+    }
+
+    /**
+     * Returns a list of auxiliary repositories for the given exercise.
+     *
+     * @param exerciseId The id of the exercise
+     * @return A list of auxiliary repositories, or an empty list if the migration service does not support auxiliary repositories
+     */
+    private List<AuxiliaryRepository> getAuxiliaryRepositories(Long exerciseId) {
+        return auxiliaryRepositoryRepository.findByExerciseId(exerciseId);
+    }
+
+    /**
+     * Migrate the solution participations. Also Migrates the test repository of the programming exercise since we have it
```

Comment:
Does the same reasoning apply for auxiliary repositories as well, as they are also migrated in this method?

Reply:
yes, I'll update the comment